### ISSUE TITLE
Bouding boxes good but slim masks

### ISSUE THREAD

**Issue by davesean** [matterport/Mask_RCNN#2024]
I have been training the mask rcnn for a while and can't seem to figure out this behaviour.
As can be seen in the figure below, the masks don't go out to the full boundary of the item.

![00000133](https://user-images.githubusercontent.com/12168969/75777820-f5c32900-5d56-11ea-9365-b054facd255c.jpg)

The masks that are being fed in during training have full masks.

Any ideas? I have played around with the parameters a bit (adding one more layer for finer masks helped a bit) but I couldn't resolve this behaviour completely.

**Comment by JordanMakesMaps** [comment]
Hey, I had a similar issue with semantic segmentation problem I was working on. I was training on down-sized images so my computer could handle the memory requirements, but then fed those same down-size images to my model during testing and had this problem.

The problem was solved by using the original sized images during testing time (e.g. don't down-size them during testing).

**Comment by davesean** [comment]
Thanks Jordan for you input!

One question about your images, were the masks just low-resolution looking or were the masks smaller than they should be like in my case?

As you said your problem was with semantic segmentation, were you using mask rcnn or another model with a U-net like architecture?
There it would make sense that you could insert larger images than during training as those networks are just made up of convolutions and deconvolutions.

**Comment by JordanMakesMaps** [comment]
The masks were actually larger than the objects they were supposed to represent (so you might actually want to try original images, smaller and larger?)

And yes, I was using a U-Net architecture so I could see the model's input size at creation to (None, None, 3).

**Comment by davesean** [comment]
After some inspection of the data, I noticed when using the inspection script that some masks would vanish and the remaining ground truth mini masks would show the same behaviour as above due to subsampling.

in the utils.py I commented out the subsampling and replaced it with another one:

`# m = skimage.transform.resize(m, mini_shape, order=1, mode="constant")`
`m = scipy.misc.imresize(m.astype(np.int), mini_shape, interp='nearest')`

using bilinear interpolation didn't give us the best results for our data.

We didn't run into this problem beforehand, because initially we were using lower quality images that didn't need to be resized as much or at all.
Now our data has become higher quality and downsampling was really a thing.
It was very strange that some masks were lost completely from the 'skimage.transform.resize()' call.

Hope this helps someone else!
